src/tests/ftest/mdtest/mdtest_small.py

```diff
@@ -58,3 +58,17 @@ def test_mdtest_small(self):
             self.execute_mdtest()
             # re-set mdtest params before next iteration
             self.mdtest_cmd.get_params(self)
+
+    def tearDown(self):
+        """ To avoid the issue from DAOS-7304, timeout when destroying a pool,
+        the plan is to delete each container within pool, this should simplify
+        the pool delete process.
+        """
+        daos_cmd = self.get_daos_command()
+        pool_uuid = self.pool.uuid
+        containers = daos_cmd.pool_list_cont(pool_uuid)
+
+        for cont in containers["uuids"]:
+            daos_cmd.container_destroy(pool_uuid, cont, force=True)
+
+        super().tearDown()
```

Comment:
To me the solution looks like we are trying to hide the bad behavior of DAOS. A customer should be able to destroy the pool with containers. Deleting the containers in the script seems that we are not testing a valid used case... Also, we are using --force option for deleting the containers... Why? Do we expect something is bad happening on the test?

Reply:
It is more of a problem with the time required to destroy the pool in the test environment. By deleting each container the pool destroy is simplified to just delete the pool metadata and not all the data within the containers which are used exhaustively by mdtest. It is a different way to the cleaning after the test run.

Reply:
@rpadma2  thoughts?

Reply:
Is it possible to write a common utility method to delete all containers in a pool rather than test specific. Still not clear why we can super.tearDown() here?

Reply:
Yes it is possible to modify the original teardown to first delete containers and then pools, but unsure we want to do it? @phender  thoughts?

Im using the force option for the same reason the pool destroy is also forced within the super teardown, which I really don't know :)

Reply:
We already delete all the containers inside each pool prior to destroying each pool - assuming they are all defined within `self.containers` and their pools are defined in `self.pools` - which is the case https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/mdtest_test_base.py#L47-L51.

Reply:
In general, this is not the recommended way to override the tearDown() method and it breaks some of the logging. The recommended way is to override the pre_tear_down() method - which should return a list of any errors that occurred during its call.